guides/keras_nlp/fnet_classification.py

```diff
@@ -0,0 +1,374 @@
+[KerasNLP](https://github.com/keras-team/keras-nlp). KerasNLP makes life easier
+for people who want to build NLP pipelines! :)
+
+### Model
+
+Transformer-based language models (LMs) such as BERT, RoBERTa, XLNet, etc. have
+demonstrated the effectiveness of the self-attention mechanism for computing 
+rich embeddings for input text. However, the self-attention mechanism is an
+expensive operation, with a time complexity of `O(n^2)`, where `n` is the number
+of tokens in the input. Hence, there has been an effort to reduce the time
+complexity of the self-attention mechanism and improve performance without
+sacrificing the quality of results.
+
+In 2020, a paper titled
+[FNet: Mixing Tokens with Fourier Transforms](https://arxiv.org/abs/2105.03824)
+replaced the self-attention layer in BERT with a simple Fourier Transform layer
+for "token mixing". This resulted in comparable accuracy and a speed-up during
+training. In particular, a couple of points from the paper stand out:
+
+1. The authors claim that FNet is 80% faster than BERT on GPUs and 70% faster
+on TPUs.
+The reason for this speed-up is two-fold:
+    a. The Fourier Transform layer is unparametrized, it does not have any parameters!
+    b. The authors use Fast Fourier Transform (FFT); this reduces the time complexity
+    from O(n^2) (in the case of self-attention) to O(n log n).
+2. FNet manages to achieve 92-97% of the accuracy of BERT on the GLUE benchmark.
+"""
+
+"""
+## Setup
+
+Before we start with the implementation, let's install the KerasNLP library, and
+import all the necessary packages.
+"""
+
+"""shell
+pip install -q keras-nlp
+"""
+import keras_nlp
+import random
+import tensorflow as tf
+import os
+
+from tensorflow import keras
+from tensorflow_text.tools.wordpiece_vocab import bert_vocab_from_dataset as bert_vocab
+
+random.seed(42)
+
+"""
+Let's also define our hyperparameters.
+"""
+BATCH_SIZE = 64
+EPOCHS = 3
+MAX_SEQUENCE_LENGTH = 512
+VOCAB_SIZE = 15000
+
+EMBED_DIM = 128
+INTERMEDIATE_DIM = 512
+
+"""
+## Loading the Dataset
+
+First, let's download the IMDB dataset and extract it.
+"""
+
+"""shell
+wget http://ai.stanford.edu/~amaas/data/sentiment/aclImdb_v1.tar.gz
+tar -xzf aclImdb_v1.tar.gz
+"""
+
+"""
+Samples are present in the form of text files. Let's inspect the structure of
+the directory.
+"""
+
+print(os.listdir("./aclImdb"))
+print(os.listdir("./aclImdb/train"))
+print(os.listdir("./aclImdb/test"))
+
+"""
+The directory contains two sub-directories: `train` and `test`. Each subdirectory
+in turn contains two folders: `pos` and `neg` for positive and negative reviews,
+respectively. Before we load the dataset, let's delete the `./aclImdb/train/unsup`
+folder since it has unlabelled samples.
+"""
+
+"""shell
+rm -rf aclImdb/train/unsup
+"""
+
+"""
+We'll use the `tf.keras.utils.text_dataset_from_directory` utility to generate
+our labelled `tf.data.Dataset` dataset from text files.
+"""
+
+train_ds = tf.keras.utils.text_dataset_from_directory(
```

Comment:
Use `from tensorflow import keras` so you don't have to use `tf.keras` everywhere